src/huggingface_hub/fastai_utils.py

```diff
@@ -351,15 +353,53 @@ def from_pretrained_fastai(
     return load_learner(os.path.join(storage_folder, "model.pkl"))
 
 
+@_deprecate_positional_args(version="0.12")
+@_deprecate_arguments(
+    version="0.12",
+    deprecated_args={
+        "repo_url",
+        "repo_path_or_name",
+        "organization",
+        "use_auth_token",
+        "git_user",
+        "git_email",
+        "skip_lfs_files",
+    },
+)
 @validate_hf_hub_args
 def push_to_hub_fastai(
     learner,
-    repo_id: str,
+    # NOTE: deprecated signature that will change in 0.12
+    repo_path_or_name: Optional[str] = None,
+    repo_url: Optional[str] = None,
     commit_message: Optional[str] = "Add model",
-    private: bool = False,
-    token: Optional[str] = None,
+    organization: Optional[str] = None,
+    private: Optional[bool] = None,
+    api_endpoint: Optional[str] = None,
+    use_auth_token: Optional[Union[bool, str]] = None,
+    git_user: Optional[str] = None,
+    git_email: Optional[str] = None,
     config: Optional[dict] = None,
-    **kwargs,
+    skip_lfs_files: bool = False,
+    # NOTE: New arguments since 0.9
+    repo_id: Optional[str] = None,  # optional only until 0.12
+    token: Optional[str] = None,
+    branch: Optional[str] = None,
+    create_pr: Optional[bool] = None,
+    allow_patterns: Optional[Union[List[str], str]] = None,
+    ignore_patterns: Optional[Union[List[str], str]] = None,
+    # TODO (release 0.12): signature must be the following
+    # repo_id: str,
+    # *,
+    # commit_message: Optional[str] = "Add model",
+    # private: Optional[bool] = None,
+    # api_endpoint: Optional[str] = None,
+    # token: Optional[str] = None,
+    # branch: Optional[str] = None,
+    # create_pr: Optional[bool] = None,
+    # config: Optional[dict] = None,
+    # allow_patterns: Optional[Union[List[str], str]] = None,
+    # ignore_patterns: Optional[Union[List[str], str]] = None,
```

Comment:
Hey @nandwalritik thank you for working on this integration :slightly_smiling_face: 

Here I think there are too much new arguments. The previous signature was:
```py
def push_to_hub_fastai(
    learner,
    repo_id: str,
    commit_message: Optional[str] = "Add model",
    private: bool = False,
    token: Optional[str] = None,
    config: Optional[dict] = None,
    **kwargs,
):
# kwargs is expecting "api_endpoint", "git_user", "git_email"
```
Actually that is very fortunate because we do not need to deprecate any of those arguments as in the other `push_to_hub` methods.

So here my suggestion would be:
- Keep the existing signature
- Add the `branch`, `create_pr`, `allow_patterns`, `ignore_patterns` arguments as you did
    - Document them in the docstring
- Add `api_endpoint` as an explicit argument (defaults to None) instead of having it as a `kwarg` + document it
- Keep `git_user` and `git_email` as arguments (default to None) but deprecate them, remove them from the docstring and do not use them in the source code. Since we are now using http-based and not git-based, those arguments are not needed anymore.
- Remove everything below the line `if repo_path_or_name is None and repo_url is None:`: those arguments were already not used so no existing codebase expect this behavior. Do need to have a git-based version anymore :) 

And I think that will be it ! I hope it's clear enough to continue working on it. Do not hesitate to ask if you have any questions.

Reply:
Also I haven't investigated it but `black` formatter seems to complain. No big deal, you install to install it (`pip install black`) and run `black ./src/huggingface_hub` and commit the changes.